File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/documents/config/RepositoriesInMemoryConfig.java
package io.zenwave360.example.clinicaltool.modules.documents.config;

import io.zenwave360.example.clinicaltool.modules.documents.*;
import io.zenwave360.example.clinicaltool.modules.documents.inmemory.*;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Primary;

// @Configuration
public class RepositoriesInMemoryConfig {

    protected final DocumentInfoRepository documentInfoRepository = new DocumentInfoRepositoryInMemory();

    @Bean
    @Primary
    public <T extends DocumentInfoRepository> T documentInfoRepository() {
        return (T) documentInfoRepository;
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/documents/config/RepositoriesInMemoryConfig.kt
package io.zenwave360.example.clinicaltool.modules.documents.config

import io.zenwave360.example.clinicaltool.modules.documents.*
import io.zenwave360.example.clinicaltool.modules.documents.inmemory.*
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Primary

// @Configuration
open class RepositoriesInMemoryConfig {

    protected val documentInfoRepository = DocumentInfoRepositoryInMemory()

    @Bean
    @Primary
    fun documentInfoRepository(): DocumentInfoRepositoryInMemory {
        return documentInfoRepository
    }
}
